### DISCUSSION TITLE
Charging from grid

### DISCUSSION THREAD

**Discussion by alexgoller** [wlcrs/huawei_solar#851] [General]
I read through the documentation, but struggle to find a way to charge from grid. I run two inverters cascaded and a Luna 10kWh battery and would love to charge the battery from grid in times of cheap electricity.

anyone got this to work or is it possible at all?

**Comment by Xexanos** [comment]
To charge from grid you need to:

1. make sure, that charge from grid is enabled in the settings from the Batteries
2. Use the action "Huawei Solar: Forcible Charge" or "Huawei Solar: Forcible Charge to a SoC level"

At least that is the way it works for me.

On another note: make sure, that charging the batterie is actually worth ist since the prices have to difer quite a bit for this beeing a viable option since there are considerable losses involved in charging the batteries from the grid.

**Reply by alexgoller** [reply]
Thanks for that, will enable that and let you know if it works. Octopus charges around 16ct / kwh, so not crazy good, but worth a try.

**Reply by tmenguy** [reply]
@Xexanos : any hint of the loss? the only data I see is that we do have our inverter efficiency, around 98% for Huawei models, so 2% of loss, is it something you have seen or is it vastly more?

**Reply by Xexanos** [reply]
The loss depends on how much load is on the Inverter. The more Energy you pull, the more efficient it gets. It is also described in the wiki of this integration (https://github.com/wlcrs/huawei_solar/wiki/Daily-Solar-Yield)
In my case the normal draw is somewhere between 300 and 500 W for my 5000 W inverter. so the efficiency is somewhere between 94% and 96%. And for roundtrip efficiency you have to multiply the values (assuming 98% since you probably charge with the full power) so in my case I land at roughly 98% * 95% = 93.1%.

**Reply by Xexanos** [reply]
If you want to automate the whole thing I am currently trying to setup [Akkudoktor EOS](https://akkudoktor-eos.readthedocs.io/en/latest/index.html) with the help of the [eos-connect Addon](https://github.com/ohAnd/EOS_connect). Maybe this could be something for you too.
